pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -497,13 +516,30 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
 
 			if currentListener != nil {
 				// merge the newly built listener with the existing listener
+				// if and only if the filter chains have distinct conditions
+				// Extract the current filter chain matches
+				// For every new filter chain match being added, check if any previous match is same
+				// if so, skip adding this filter chain with a warning
+				// This is very unoptimized.
 				newFilterChains := make([]listener.FilterChain, 0, len(currentListener.FilterChains)+len(mutable.Listener.FilterChains))
 				newFilterChains = append(newFilterChains, currentListener.FilterChains...)
-				newFilterChains = append(newFilterChains, mutable.Listener.FilterChains...)
+				for _, incomingFilterChain := range mutable.Listener.FilterChains {
```

Comment:
what is the difference between this and the existing logic?

Reply:
I think this one finds the issues immediately vs the existing logic does it in the end. Also, this one compares against sni and destination CIDRs. But yeah, they are the same. I found the other logic after I added this :)